weaver/cli.py

```diff
@@ -0,0 +1,717 @@
+        base = self._get_url(url)
+        result = self.describe(process_id, url=base)
+        if not result.success:
+            return result
+        outputs = result.body.get("outputs")
+        for output_id in outputs:
+            # use 'value' to have all outputs reported in body as 'value/href' rather than 'Link' headers
+            data["outputs"][output_id] = {"transmissionMode": EXECUTE_TRANSMISSION_MODE_VALUE}
+
+        LOGGER.info("Executing [%s] with inputs:\n%s", process_id, self._json2text(inputs))
+        path = f"{base}/processes/{process_id}/execution"  # use OGC-API compliant endpoint (not '/jobs')
+        resp = request_extra("POST", path, json=data, headers=self._headers, settings=self._settings)
+        result = self._parse_result(resp)
+        if not monitor or not result.success:
+            return result
+        # although Weaver returns "jobID" in the body for convenience,
+        # employ the "Location" header to be OGC-API compliant
+        job_url = resp.headers.get("Location", "")
+        time.sleep(1)  # small delay to ensure process execution had a chance to start before monitoring
+        return self.monitor(job_url, timeout=timeout)
+
+    def status(self, job_reference, url=None):
+        """
+        Obtain the status of a :term:`Job`.
+
+        :param job_reference: Either the full :term:`Job` status URL or only its UUID.
+        :param url: Instance URL if not already provided during client creation.
+        :returns: retrieved status of the job.
+        """
+        job_id, job_url = self._parse_job_ref(job_reference, url)
+        LOGGER.info("Getting job status: [%s]", job_id)
+        resp = request_extra("GET", job_url, headers=self._headers)
+        return self._parse_result(resp)
+
+    def monitor(self, job_reference, timeout=None, delta=None, wait_for_status=STATUS_SUCCEEDED, url=None):
+        # type: (str, Optional[int], Optional[int], str, Optional[str]) -> OperationResult
+        """
+        Monitor the execution of a :term:`Job` until completion.
+
+        :param job_reference: Either the full :term:`Job` status URL or only its UUID.
+        :param timeout: timeout (seconds) of monitoring until completion or abort.
+        :param delta: interval (seconds) between polling monitor requests.
+        :param wait_for_status: monitor until the requested status is reached (default: job failed or succeeded).
+        :param url: Instance URL if not already provided during client creation.
+        :return: result of the successful or failed job, or timeout of monitoring process.
+        """
+        job_id, job_url = self._parse_job_ref(job_reference, url)
+        remain = timeout = timeout or self.monitor_timeout
+        delta = delta or self.monitor_delta
+        LOGGER.info("Monitoring job [%s] for %ss at intervals of %ss.", job_id, timeout, delta)
+        once = True
+        body = None
+        while remain >= 0 or once:
```

Comment:
Getting an error here, if I use the `-T` argument.
command used : `weaver monitor -u http://localhost:4001 -j d0916f60-62be-4a45-8f5b-67d79a89c45f -T 300`

Error output :
```
Traceback (most recent call last):
  File "/home/cumminch/.conda/envs/weaver/bin/weaver", line 33, in <module>
    sys.exit(load_entry_point('weaver', 'console_scripts', 'weaver')())
  File "/home/cumminch/weaver/weaver/cli.py", line 707, in main
    result = getattr(client, oper)(**kwargs)
  File "/home/cumminch/weaver/weaver/cli.py", line 426, in monitor
    while remain >= 0 or once:
TypeError: '>=' not supported between instances of 'str' and 'int'
```